### PR TITLE
Brazilian portuguese translation

### PR DESCRIPTION
**Pull request by atilacamurca**

### CODE REVIEW

**Comment by atilacamurca** [comment]
I did the translation using the web tool <https://crowdin.com/project/mgba-pt-br>.

I think it would be interest if the mgba project had its own page. It's free for open source projects and it's easy to get started. You just create a project page and submit the `.ts` files.

**Comment by endrift** [comment]
I've been trying to decide between Crowdin, Transifex and Weblate before committing to one platform, but I have been meaning to use one.

I'm going to wait for someone I know who knows pt_BR to review this before I merge it though.

**Comment by atilacamurca** [comment]
Great! Thanks for the response. I understand we have to be careful with those type of contributions.

Two things I want to ask is:

- To have a template `.ts` file for those who are starting to translate. That is, an  empty file with only the english string. The crowdin tool was able to do that for me, but it would be nice for those using other tools
- That the file have guiding comments for more context. For example, I saw in other translations that people translate the word **English** instead of putting his own language. I could see that based on other translations that I did, but it was not clear.

**Comment by lotharsm** [comment]
@endrift somewhat unrelated - but if you decide to use a translation system, I'd always prefer Weblate over Transifex (I simply can't stand Transifex, it feels so clumsy...)

File: src/platform/qt/ts/mgba-pt_BR.ts [hunk 1]
```diff
@@ -0,0 +1,5445 @@
...
+      <source>Graphics Interchange Format (*.gif);;Animated Portable Network Graphics (*.png *.webp *.apng)</source>
+      <translation>Graphics Interchange Format (*.gif);;Animated Portable Network Graphics (*.png *.webp *.apng)</translation>
+    </message>
+  </context>
+  <context>
+    <name>QGBA::IOViewer</name>
+    <message>
+      <location filename="../IOViewer.cpp" line="30"/>
+      <source>Background mode</source>
+      <translation>Modo plano de fundo</translation>
```
Comment (yuriks) [inline src/platform/qt/ts/mgba-pt_BR.ts:1664]: "Modo dos planos de fundo"

File: src/platform/qt/ts/mgba-pt_BR.ts [hunk 2]
```diff
@@ -0,0 +1,5445 @@
...
+    </message>
+    <message>
+      <location filename="../IOViewer.cpp" line="31"/>
+      <source>Mode 0: 4 tile layers</source>
+      <translation>Modo 0: 4 camadas de bloco</translation>
+    </message>
+    <message>
+      <location filename="../IOViewer.cpp" line="32"/>
+      <source>Mode 1: 2 tile layers + 1 rotated/scaled tile layer</source>
+      <translation>Modo 1: 2 camadas de blocos + 1 camada de bloco rotacionado/dimensionado</translation>
```
Comment (yuriks) [inline src/platform/qt/ts/mgba-pt_BR.ts:1674]: "rotacionado/escalonado" pra ser consistente com a linha abaixo.

File: src/platform/qt/ts/mgba-pt_BR.ts [hunk 3]
```diff
@@ -0,0 +1,5445 @@
...
+    </message>
+    <message>
+      <location filename="../IOViewer.cpp" line="43"/>
+      <source>Linear OBJ tile mapping</source>
+      <translation>Mapeamento linear do OBJ</translation>
+    </message>
+    <message>
+      <location filename="../IOViewer.cpp" line="44"/>
+      <source>Force blank screen</source>
+      <translation>Forçar tela em branco</translation>
```
Comment (yuriks) [inline src/platform/qt/ts/mgba-pt_BR.ts:1719]: "Forçar limpeza da tela"

File: src/platform/qt/ts/mgba-pt_BR.ts [hunk 4]
```diff
@@ -0,0 +1,5445 @@
...
+    </message>
+    <message>
+      <location filename="../IOViewer.cpp" line="63"/>
+      <source>Enable VCounter IRQ generation</source>
+      <translation>Habilitar geração de VCounter IRQ</translation>
+    </message>
+    <message>
+      <location filename="../IOViewer.cpp" line="64"/>
+      <source>VCounter scanline</source>
+      <translation>VCounter scanline</translation>
```
Comment (yuriks) [inline src/platform/qt/ts/mgba-pt_BR.ts:1794]: "Scanline do VCounter"

File: src/platform/qt/ts/mgba-pt_BR.ts [hunk 5]
```diff
@@ -0,0 +1,5445 @@
...
+      <source>Integer part</source>
+      <translation>Parte inteira</translation>
+    </message>
+    <message>
+      <location filename="../IOViewer.cpp" line="163"/>
+      <location filename="../IOViewer.cpp" line="172"/>
+      <location filename="../IOViewer.cpp" line="201"/>
+      <location filename="../IOViewer.cpp" line="210"/>
+      <source>Integer part (bottom)</source>
+      <translation>Parte Inteira (inferior)</translation>
```
Comment (yuriks) [inline src/platform/qt/ts/mgba-pt_BR.ts:1905]: "inteira" minúscula pra ser consistente com as outras linhas.

File: src/platform/qt/ts/mgba-pt_BR.ts [hunk 6]
```diff
@@ -0,0 +1,5445 @@
...
+      <location filename="../IOViewer.cpp" line="205"/>
+      <location filename="../IOViewer.cpp" line="214"/>
+      <source>Integer part (top)</source>
+      <translation>Parte inteira (superior)</translation>
+    </message>
+    <message>
+      <location filename="../IOViewer.cpp" line="218"/>
+      <location filename="../IOViewer.cpp" line="223"/>
+      <source>End x</source>
+      <translation>Final x</translation>
```
Comment (yuriks) [inline src/platform/qt/ts/mgba-pt_BR.ts:1919]: "X final", "X inicial", etc.

File: src/platform/qt/ts/mgba-pt_BR.ts [hunk 7]
```diff
@@ -0,0 +1,5445 @@
...
+    </message>
+    <message>
+      <location filename="../IOViewer.cpp" line="269"/>
+      <source>Background mosaic size horizontal</source>
+      <translation>Tamanho horizontal do mosaico de fundo</translation>
+    </message>
+    <message>
+      <location filename="../IOViewer.cpp" line="270"/>
+      <source>Object mosaic size vertical</source>
+      <translation>Tamanho do mosaico do objeto vertical</translation>
```
Comment (yuriks) [inline src/platform/qt/ts/mgba-pt_BR.ts:2072]: "Tamanho vertical do mosaico de objeto", pra ser consistente. (Linha abaixo também.)

File: src/platform/qt/ts/mgba-pt_BR.ts [hunk 8]
```diff
@@ -0,0 +1,5445 @@
...
+    </message>
+    <message>
+      <location filename="../IOViewer.cpp" line="285"/>
+      <source>Additive blending</source>
+      <translation>Blending aditivo</translation>
+    </message>
+    <message>
+      <location filename="../IOViewer.cpp" line="286"/>
+      <source>Brighten</source>
+      <translation>Brilhante</translation>
```
Comment (yuriks) [inline src/platform/qt/ts/mgba-pt_BR.ts:2127]: "Clarear"

File: src/platform/qt/ts/mgba-pt_BR.ts [hunk 9]
```diff
@@ -0,0 +1,5445 @@
...
+    </message>
+    <message>
+      <location filename="../IOViewer.cpp" line="287"/>
+      <source>Darken</source>
+      <translation>Escurecido</translation>
```
Comment (yuriks) [inline src/platform/qt/ts/mgba-pt_BR.ts:2132]: "Escurecer"

File: src/platform/qt/ts/mgba-pt_BR.ts [hunk 10]
```diff
@@ -0,0 +1,5445 @@
...
+    </message>
+    <message>
+      <location filename="../LogConfigModel.cpp" line="91"/>
+      <source>Stub</source>
+      <translation>Stub</translation>
+    </message>
+    <message>
+      <location filename="../LogConfigModel.cpp" line="93"/>
+      <source>Game Error</source>
+      <translation>Erro no Jogo</translation>
```
Comment (yuriks) [inline src/platform/qt/ts/mgba-pt_BR.ts:3120]: "Erro do Jogo"

File: src/platform/qt/ts/mgba-pt_BR.ts [hunk 11]
```diff
@@ -0,0 +1,5445 @@
...
+    <message>
+      <location filename="../SettingsView.cpp" line="328"/>
+      <source>Shortcuts</source>
+      <translation>Atalhos</translation>
+    </message>
+    <message>
+      <location filename="../SettingsView.cpp" line="340"/>
+      <location filename="../SettingsView.cpp" line="350"/>
+      <source>Shaders</source>
+      <translation>Sombreados</translation>
```
Comment (yuriks) [inline src/platform/qt/ts/mgba-pt_BR.ts:3526]: Acho que não se traduz "Shader", mas senão, acho que "Filtros" seria mais adequado.

File: src/platform/qt/ts/mgba-pt_BR.ts [hunk 12]
```diff
@@ -0,0 +1,5445 @@
...
+      <source>(%1×%2)</source>
+      <translation>(%1×%2)</translation>
+    </message>
+  </context>
+  <context>
+    <name>QGBA::ShaderSelector</name>
+    <message>
+      <location filename="../ShaderSelector.cpp" line="52"/>
+      <source>No shader active</source>
+      <translation>Nenhum sombreamento ativo</translation>
```
Comment (yuriks) [inline src/platform/qt/ts/mgba-pt_BR.ts:3544]: etc. Acho que "sombreamento" fica confuso.

File: src/platform/qt/ts/mgba-pt_BR.ts [hunk 13]
```diff
@@ -0,0 +1,5445 @@
...
+      <source>Export tile</source>
+      <translation>Exportar bloco</translation>
+    </message>
+  </context>
+  <context>
+    <name>QGBA::VideoView</name>
+    <message>
+      <location filename="../VideoView.cpp" line="196"/>
+      <source>Failed to open output video file: %1</source>
+      <translation>Falha ao abrir arquivo de saída de vídeo: %1</translation>
```
Comment (yuriks) [inline src/platform/qt/ts/mgba-pt_BR.ts:3614]: "arquivo de vídeo de saída"

File: src/platform/qt/ts/mgba-pt_BR.ts [hunk 14]
```diff
@@ -0,0 +1,5445 @@
...
+    </message>
+    <message>
+      <location filename="../Window.cpp" line="1126"/>
+      <source>Load &amp;patch...</source>
+      <translation>Carregar &amp;patch...</translation>
+    </message>
+    <message>
+      <location filename="../Window.cpp" line="1129"/>
+      <source>Boot BIOS</source>
+      <translation>Iniciar BIOS</translation>
```
Comment (yuriks) [inline src/platform/qt/ts/mgba-pt_BR.ts:3837]: "Rodar BIOS" acho que fica melhor

File: src/platform/qt/ts/mgba-pt_BR.ts [hunk 15]
```diff
@@ -0,0 +1,5445 @@
...
+    </message>
+    <message>
+      <location filename="../Window.cpp" line="1241"/>
+      <source>Sh&amp;utdown</source>
+      <translation>&amp;Desligar</translation>
+    </message>
+    <message>
+      <location filename="../Window.cpp" line="1245"/>
+      <source>Yank game pak</source>
+      <translation>Yank game pak</translation>
```
Comment (yuriks) [inline src/platform/qt/ts/mgba-pt_BR.ts:3963]: "Remover game pak"

File: src/platform/qt/ts/mgba-pt_BR.ts [hunk 16]
```diff
@@ -0,0 +1,5445 @@
...
+    </message>
+    <message>
+      <location filename="../Window.cpp" line="1381"/>
+      <source>Toggle fullscreen</source>
+      <translation>Alternar tela cheia</translation>
+    </message>
+    <message>
+      <location filename="../Window.cpp" line="1384"/>
+      <source>Lock aspect ratio</source>
+      <translation>Bloquear proporção</translation>
```
Comment (yuriks) [inline src/platform/qt/ts/mgba-pt_BR.ts:4073]: "Fixar proporção" ou "Travar proporção"

File: src/platform/qt/ts/mgba-pt_BR.ts [hunk 17]
```diff
@@ -0,0 +1,5445 @@
...
+    </message>
+    <message>
+      <location filename="../SensorView.ui" line="47"/>
+      <source>System time</source>
+      <translation>Horário do sistema</translation>
+    </message>
+    <message>
+      <location filename="../SensorView.ui" line="60"/>
+      <source>Start time at</source>
+      <translation>Horário de início em</translation>
```
Comment (yuriks) [inline src/platform/qt/ts/mgba-pt_BR.ts:4460]: "Horário iniciando em"

File: src/platform/qt/ts/mgba-pt_BR.ts [hunk 18]
```diff
@@ -0,0 +1,5445 @@
...
+    </message>
+    <message>
+      <location filename="../SensorView.ui" line="70"/>
+      <source>Now</source>
+      <translation>Agora</translation>
+    </message>
+    <message>
+      <location filename="../SensorView.ui" line="97"/>
+      <source>MM/dd/yy hh:mm:ss AP</source>
+      <translation>MM/dd/yy hh:mm:ss AP</translation>
```
Comment (yuriks) [inline src/platform/qt/ts/mgba-pt_BR.ts:4470]: `dd/MM/yy hh:mm:ss` pra ficar no formato brasileiro.

File: src/platform/qt/ts/mgba-pt_BR.ts [hunk 19]
```diff
@@ -0,0 +1,5445 @@
...
+    <message>
+      <location filename="../SettingsView.ui" line="270"/>
+      <location filename="../SettingsView.ui" line="310"/>
+      <source>Mute</source>
+      <translation>Mudo</translation>
+    </message>
+    <message>
+      <location filename="../SettingsView.ui" line="279"/>
+      <source>Fast forward volume:</source>
+      <translation>Avançar volume:</translation>
```
Comment (yuriks) [inline src/platform/qt/ts/mgba-pt_BR.ts:4653]: "Volume durante avanço rápido:"

File: src/platform/qt/ts/mgba-pt_BR.ts [hunk 20]
```diff
@@ -0,0 +1,5445 @@
...
+    </message>
+    <message>
+      <location filename="../SettingsView.ui" line="326"/>
+      <source>Display driver:</source>
+      <translation>Exibir driver:</translation>
```
Comment (yuriks) [inline src/platform/qt/ts/mgba-pt_BR.ts:4658]: "Driver de exibição:"

File: src/platform/qt/ts/mgba-pt_BR.ts [hunk 21]
```diff
@@ -0,0 +1,5445 @@
...
+    </message>
+    <message>
+      <location filename="../SettingsView.ui" line="428"/>
+      <source>Audio</source>
+      <translation>Áudio</translation>
+    </message>
+    <message>
+      <location filename="../SettingsView.ui" line="437"/>
+      <source>Lock aspect ratio</source>
+      <translation>Bloquear proporção</translation>
```
Comment (yuriks) [inline src/platform/qt/ts/mgba-pt_BR.ts:4704]: Novamente, "fixar" ou "travar" fica melhor eu acho.

File: src/platform/qt/ts/mgba-pt_BR.ts [hunk 22]
```diff
@@ -0,0 +1,5445 @@
...
+    </message>
+    <message>
+      <location filename="../SettingsView.ui" line="837"/>
+      <source>Run all</source>
+      <translation>Executar todos</translation>
+    </message>
+    <message>
+      <location filename="../SettingsView.ui" line="842"/>
+      <source>Remove known</source>
+      <translation>Remover conhecido</translation>
```
Comment (yuriks) [inline src/platform/qt/ts/mgba-pt_BR.ts:4912]: "conhecido**s**"

File: src/platform/qt/ts/mgba-pt_BR.ts [hunk 23]
```diff
@@ -0,0 +1,5445 @@
...
+    </message>
+    <message>
+      <location filename="../SettingsView.ui" line="1150"/>
+      <source>SGB BIOS file:</source>
+      <translation>Arquivo SGB BIOS:</translation>
+    </message>
+    <message>
+      <location filename="../SettingsView.ui" line="1185"/>
+      <source>Save games</source>
+      <translation>Salvar jogos</translation>
```
Comment (yuriks) [inline src/platform/qt/ts/mgba-pt_BR.ts:5033]: "Arquivos de salvamento"

File: src/platform/qt/ts/mgba-pt_BR.ts [hunk 24]
```diff
@@ -0,0 +1,5445 @@
...
+      <location filename="../SettingsView.ui" line="1496"/>
+      <location filename="../SettingsView.ui" line="1532"/>
+      <location filename="../SettingsView.ui" line="1568"/>
+      <source>Game Boy Advance (AGB)</source>
+      <translation>Game Boy Advance (AGB)</translation>
+    </message>
+    <message>
+      <location filename="../SettingsView.ui" line="1583"/>
+      <source>Default BG colors:</source>
+      <translation>Cores padrão BG:</translation>
```
Comment (yuriks) [inline src/platform/qt/ts/mgba-pt_BR.ts:5102]: "Cores BG padrão:" ou "Cores padrão do BG:"

File: src/platform/qt/ts/mgba-pt_BR.ts [hunk 25]
```diff
@@ -0,0 +1,5445 @@
...
+      <source>Default sprite colors 2:</source>
+      <translation>Cores padrão do sprite 2:</translation>
+    </message>
+  </context>
+  <context>
+    <name>ShaderSelector</name>
+    <message>
+      <location filename="../ShaderSelector.ui" line="14"/>
+      <source>Shaders</source>
+      <translation>Sombreados</translation>
```
Comment (yuriks) [inline src/platform/qt/ts/mgba-pt_BR.ts:5130]: "Filtros" etc.

File: src/platform/qt/ts/mgba-pt_BR.ts [hunk 26]
```diff
@@ -0,0 +1,5445 @@
...
+    </message>
+    <message>
+      <location filename="../VideoView.ui" line="340"/>
+      <source>AAC</source>
+      <translation>AAC</translation>
+    </message>
+    <message>
+      <location filename="../VideoView.ui" line="345"/>
+      <source>Uncompressed</source>
+      <translation>Descompactado</translation>
```
Comment (yuriks) [inline src/platform/qt/ts/mgba-pt_BR.ts:5402]: "Sem compressão"

**Review by yuriks — CHANGES_REQUESTED** [review]
I found a few lines that sound incorrect in context, so left suggestions for those. I also left suggestions for several other things where I thought the wording or word choice could be improved.

Thanks for doing the translation!

**Comment by yuriks** [comment]
(Note: I marked my review "Request changes" but it's up to @endrift whether to merge or not. Most of it looks fine as-is.)

File: src/platform/qt/ts/mgba-pt_BR.ts [hunk 27]
```diff
@@ -0,0 +1,5445 @@
...
+    </message>
+    <message>
+      <location filename="../MemoryDump.ui" line="76"/>
+      <source>Byte Count:</source>
+      <translation>Número de bytes:</translation>
+    </message>
+    <message>
+      <location filename="../MemoryDump.ui" line="108"/>
+      <source>Dump across banks</source>
+      <translation>Dump across banks</translation>
```
[the remark this answers is not in the document]
Comment (atilacamurca) [reply src/platform/qt/ts/mgba-pt_BR.ts:663]: Obrigado @yuriks pelas sugestões. Algumas string eu deixei de propósito sem tradução devido a dificuldade de entendimento por parte do usuário. Vou revisar os arquivos com as sugestões e logo darei novo retorno.

File: src/platform/qt/ts/mgba-pt_BR.ts (hunk 12 above)
Comment (atilacamurca) [reply src/platform/qt/ts/mgba-pt_BR.ts:3544]: @yuriks o que você acha de não traduzirmos shader? Em caso positivo posso trocar logo todas as ocorrências.

Comment (yuriks) [reply src/platform/qt/ts/mgba-pt_BR.ts:3544]: Acho válido, e provavelmente uma boa ideia pra ser menos ambíguo. Pra falar a verdade não costumo usar software traduzido pra Português muito, então não tinha certeza do jeito comum de traduzir "shader" ou se geralmente é só mantido cono palavra estrangeira.

File: src/platform/qt/ts/mgba-pt_BR.ts [hunk 28]
```diff
@@ -0,0 +1,5445 @@
...
+    </message>
+    <message>
+      <location filename="../OverrideView.ui" line="359"/>
+      <source>Wisdom Tree (Unlicensed)</source>
+      <translation>Wisdom Tree (Não licenciado)</translation>
+    </message>
+    <message>
+      <location filename="../OverrideView.ui" line="364"/>
+      <source>Pokémon Jade/Diamond (Unlicensed)</source>
+      <translation>Pokémon Jade/Diamond (Sem licença)</translation>
```
Comment (yuriks) [inline src/platform/qt/ts/mgba-pt_BR.ts:1266]: This line should be updated with "Não licenciado" to match the one before.

**Review by yuriks — APPROVED** [review]
Really sorry for taking a while to get back to this. I spotted one line that wasn't updated, but other than that it looks good to merge to me!

File: src/platform/qt/ts/mgba-pt_BR.ts (hunk 28 above)
Comment (atilacamurca) [reply src/platform/qt/ts/mgba-pt_BR.ts]: I just updated this translation, thank you.

**Review by yuriks — APPROVED** [review]

**Comment by atilacamurca** [comment]
Thank you @yuriks for the review and suggestions.

And thank you @endrift and all contributors of the project.

**Pull request merged**